### ISSUE TITLE
Need help understanding equation (3) and (4)

### ISSUE THREAD

**Issue by linzhiqiu** [frank-xwang/debiased-pseudo-labeling#3]
Thank you for such a great paper on semi-supervised learning!

I would like to ask for more reference on Controlled Direct Effect and how this method is related to your equation (3) and (4) in main paper.

Without looking into the casual inference math, it seems eq (3) and (4) is simply adjusting the pseud-label labels by averaged class probabilities of all unlabeled samples (estimated by moving average).

However, instead of adjusting probability values, you choose to adjust the raw logit values with the log probability. I am having hard time understanding it -- is it just the CDE method? 

Have you tried simply adjusting the probability values (with average prob) instead of the raw logits (with log average prob)? This to me seems more natural and I guess they will produce similar results as long as the hyperparameter is chosen wisely.

**Comment by frank-xwang** [comment]
Hi, thanks for your interest in our work! Since the scale of logits is unstable, most notably at the early training stage, we use the probability distribution pk rather than directly using the logit vector in the second term of Eqn. 3. A log function is applied to rescale \hat{p} to match the magnitude of logit. We tried to use logits directly, the performance is worse than using probabilities.

**Comment by linzhiqiu** [comment]
Thanks for the helpful answer! I was wondering if there is any recommendation on how to select the tau and qhat_m parameters. On my own long-tailed semi-supervised datasets (i-naturalist), I found debiased PL to perform worse than naive fix match, especially when I used a high confidence threshold (0.95) to filter. I don't know if this is because I didn't tune the parameters right.

**Comment by frank-xwang** [comment]
We did not try DebiasPL on iNaturalist, but DebiasPL should perform better when training long-tailed data. Do you use FixMatch with EMAN or vanilla FixMatch? If you use FixMatch with EMAN, tau=0.5 or 0.4 will usually perform well. For confidence scores thresholds, it is more difficult to achieve high confidence scores because the de-biasing module imposes a greater penalty on the many-shot classes. This may significantly reduce the speed of convergence if the confidence score is very high. Therefore, you may want to tune the confidence score thresholds. 0.95 is pretty high, I think using a lower one maybe helpful. You can check the value of the mask in the log to track the percentage of samples passing your confidence score threshold. Based on my experience on ImageNet, the final mask value should be higher than 70%. Hope it helps!

**Comment by linzhiqiu** [comment]
Thanks! I will try it out!

**Comment by frank-xwang** [comment]
Nice! I'm closing this issue for now. Please feel free to reopen it if you have more questions!
